src/site/content/en/metrics/custom-metrics/index.md

```diff
@@ -0,0 +1,377 @@
+---
+layout: post
+title: Custom metrics
+authors:
+  - philipwalton
+date: 2019-11-07
+description: |
+  Custom metrics allow you to measure and optimize aspects of your site's
+  experience that are unique to your site.
+---
+
+There's a lot of value in having [user-centric metrics](/user-centric-metrics/)
+that you can measure, universally, on any given website. These metrics allow you
+to:
+
+- Understand how real users experience the web as a whole
+- Easily compare your site to a competitor's
+- Track useful and actionable data in your analytics tools without needing to
+  write custom code
+
+Universal metrics offer a good baseline, but in many cases you need to measure
+_more_ than just these metrics in order to capture the full experience for your
+particular site.
+
+Custom metrics allow you to measure aspects of your site's experience that may
+only apply to your site, such as:
+
+- How long it takes for a single page app (SPA) to transition from one "page" to
+  another
+- How long it takes for a page to display data fetched from a database for
+  logged-in users
+- How long it takes for a server-side-rendered (SSR) app to
+  [hydrate](https://addyosmani.com/blog/rehydration/)
+- The cache hit rate for resources loaded by returning visitors
+- The event latency of click or keyboard events in a game
+
+## APIs to measure custom metrics
+
+Historically web developers haven't had many low-level APIs to measure
+performance, and as a result they've had to resort to hacks in order to measure
+whether a site was performing well.
+
+For example, it's possible to determine whether the main thread is blocked due
+to long-running JavaScript tasks by running a `requestAnimationFrame` loop and
+calculating the delta between each frame. If the delta is significantly longer
+than the display's framerate, you can report that as a long task. Such hacks are
+not recommended, though, because they actually affect performance themselves
+(e.g. drain battery).
```

Comment:
When we're happy with all the content I'll do a copyedit to remove this `e.g.` and the others I didn't catch. Leave this comment unresolved for me.